tests/test_weaver_snapshot.py

```
@@ -0,0 +1,203 @@
+    )
+    assert normalized["styleDiff"] == {"color": "rgba(1, 2, 3, 1.000)"}
+
+
+def test_animated_opacity_is_ignored_but_static_opacity_is_not() -> None:
+    """Only a node mid-animation should have its sampled opacity discarded."""
+    animated = weaver_snapshot._normalize(
+        _node(**{"animation-name": "pulse", "opacity": "0.694981"})
+    )
+    assert "opacity" not in animated["styleDiff"]
+
+    static = weaver_snapshot._normalize(_node(**{"opacity": "0.5"}))
```

Ruff (0.16.1)

[warning] Unnecessary `dict` kwargs

Remove unnecessary kwargs

(PIE804)